components/brave_wallet/browser/simple_hash_client.cc

```diff
@@ -1131,10 +1131,14 @@ SimpleHashClient::ParseMetadatas(const base::Value::Dict& dict) {
     const std::string* image = nft->FindString("image_url");
     if (image) {
       GURL original_url(*image);
-      GURL::Replacements replacements;
-      replacements.SetHostStr(kSimpleHashCdnBraveProxyHost);
-      GURL proxy_url = original_url.ReplaceComponents(replacements);
-      nft_metadata->image = proxy_url.spec();
+      if (original_url.host() == "cdn.simplehash.com") {
```

Comment:
Let's have a constant close to `kSimpleHashCdnBraveProxyHost`

Reply:
And a unit test for this replacement

Reply:
https://github.com/brave/brave-core/pull/28768/commits/782413897718cada9029ca00f60d96d2f778b18b